### ISSUE TITLE
🧪 Unit Tests: neuro-divergent-models crate

### ISSUE THREAD

**Issue by mikkihugo** [mikkihugo/claude-code-zen#231]
## 📋 Summary

Implement comprehensive unit tests for the neuro-divergent-models crate, which contains neural network model implementations (RNN, MLP, transformers, etc.).

### Models to Test:
- [ ] **Basic Models**
  - [ ] MLP (Multi-Layer Perceptron)
  - [ ] DLinear
  - [ ] NLinear
  - [ ] MLPMultivariate

- [ ] **Recurrent Models**
  - [ ] RNN (Recurrent Neural Network)
  - [ ] LSTM (Long Short-Term Memory)
  - [ ] GRU (Gated Recurrent Unit)

- [ ] **Advanced Models**
  - [ ] NBEATS
  - [ ] NBEATSx
  - [ ] NHITS
  - [ ] TiDE

- [ ] **Transformer Models**
  - [ ] TFT (Temporal Fusion Transformer)
  - [ ] Informer
  - [ ] AutoFormer
  - [ ] FedFormer
  - [ ] PatchTST
  - [ ] iTransformer

### 1. Model Construction Tests
```rust
#[cfg(test)]
mod model_tests {
    // Test model creation with various configurations
    // Test invalid configurations
    // Test parameter initialization
}
```

### 2. Forward Pass Tests
- Input dimension validation
- Output shape verification
- Numerical stability
- Edge cases (empty input, single sample)

### 3. Training Tests
- Gradient computation
- Parameter updates
- Loss convergence
- Training with different optimizers

### 4. Model-Specific Tests
- RNN: Hidden state management
- LSTM: Cell state and gates
- Transformers: Attention mechanisms
- NBEATS: Block stacking

### 5. Integration Tests with ruv-FANN
- Network conversion
- Training algorithm compatibility
- Activation function mapping

## 📊 Success Criteria
- [ ] All model types have comprehensive tests
- [ ] Forward/backward pass validation
- [ ] Training convergence tests
- [ ] Performance benchmarks per model

## 🔗 Parent Issue
Part of #229
